test/e2e/snaps/test-snap-siginsights.spec.js

```diff
@@ -94,10 +81,7 @@ describe('Test Snap Signature Insights', function () {
         await switchToNotificationWindow(driver, 4);
 
         // wait for and click sign
-        await driver.waitForSelector({
-          text: 'Sign',
-          tag: 'button',
-        });
+        await driver.delay(regularDelayMs);
```

Comment:
Nit: Could you include a comment to explain in more detail why we need a delay here? And likewise for each of the other delays. Or maybe we can refactor this into a helper function that explains in its description why we need to pause.

Ideally we'd rely less on delays because they are unreliable solutions. I think it's justified in this case as a short term strategy to reduce flakiness, but adding more delays might encourage others to do the same if we don't clearly outline that this case is a temporary exception.

Reply:
thank you @Gudahtt that's a great suggestion!! I agree that adding delays shouldn't be the preferred option, but the last alternative.
I've seen that we already had a `clickSignOnSignatureConfirmation` function in the helpers, so I've added the logic and explanation there. Let me know if this looks good!

Reply:
Looks great, thanks!